source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst

```diff
@@ -728,6 +730,19 @@ eagerly so that any operations waiting on `Connections <#connection>`_ can retry
 as soon as possible. The pool MUST NOT rely on WaitQueueTimeoutMS to clear
 requests from the WaitQueue.
 
+The clearing method MUST provide the option to close any in-use connections as part
+of the clearing (henceforth referred to as the closeInUseConnections flag in this
+specification). The closing of these connections MUST NOT be done eagerly, and
```

Comment:
> The closing of these connections MUST NOT be done eagerly

Why? In PyMongo we already close idle connections eagerly and I imagine we would do the same for in use connections too.

Reply:
The intention with this requirement is to avoid blocking the connection pool. Per https://unix.stackexchange.com/questions/145526/can-close-block, it looks like that only happens if `SO_LINGER` is set on the socket though, so maybe this requirement isn't necessary. Perhaps it should be stated more directly as:

>The closing of these connections MUST NOT block the pool or prevent it from processing further requests.

Note that we're also discussing some terminology changes here to make this "The interruption of these connections...", with the actual "closing" being delegated to checkIn.

Reply:
Ooo yeah I like that more direct phrasing better. It makes the intention clear.

Reply:
Cool, we'll update it to that then, cc @DmitryLukyanov